include/LightGBM/c_api.h

```diff
@@ -572,6 +572,9 @@ LIGHTGBM_C_EXPORT int LGBM_BoosterRefit(BoosterHandle handle,
 /*!
  * \brief Update the model by specifying gradient and Hessian directly
  *        (this can be used to support customized loss functions).
+ * \note
+ * The length of the arrays referenced by ``grad`` and ``hess`` must be equal to
+ * ``num_class * num_train_data``, this is not verified by the library, the caller must ensure this.
```

Comment:
Should this be `this IS verified by the library`, or simply delete this last sentence? Because we are actually verifying this through this pull request.

Reply:
This is the C-API docs -- the context here is the `lightgbm.so` library, rather than python or R libraries. What we're saying is that a caller to this function `LGBM_BoosterUpdateOneIterCustom` is responsible to ensure that the condition is met. I'm happy to tweak the wording, but what the python library does as a convenience to the user is not applicable here.